cli/daemon/apps/apps.go

```diff
@@ -0,0 +1,140 @@
+package apps
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/rs/zerolog/log"
+
+	"encr.dev/cli/daemon/internal/manifest"
+	"encr.dev/cli/internal/appfile"
+)
+
+var ErrNotFound = errors.New("app not found")
+
+// Instance describes an app instance known by the Encore daemon.
+type Instance struct {
+	root       string
+	localID    string
+	platformID string
+}
+
+func NewInstance(root, localID, platformID string) *Instance {
+	return &Instance{root, localID, platformID}
+}
+
+// Root returns the filesystem path for the app root.
+// It always returns a non-empty string.
+func (i *Instance) Root() string { return i.root }
+
+// LocalID reports a local, random id unique for this app,
+// as persisted in the .encore/manifest.json file.
+// It always returns a non-empty string.
+func (i *Instance) LocalID() string { return i.localID }
+
+// PlatformID reports the Encore Platform's ID for this app.
+// If the app is not linked it reports the empty string.
+func (i *Instance) PlatformID() string { return i.platformID }
+
+// PlatformOrLocalID reports PlatformID() if set and otherwise LocalID().
+func (i *Instance) PlatformOrLocalID() string {
+	if i.platformID != "" {
+		return i.platformID
+	}
+	return i.localID
+}
+
+func NewManager(db *sql.DB) *Manager {
+	return &Manager{db: db, instances: make(map[string]*Instance)}
+}
+
+// Manager keeps track of known apps and watches them for changes.
+type Manager struct {
+	db *sql.DB
+
+	mu        sync.Mutex
+	instances map[string]*Instance // root -> instance
+}
+
+// Track begins tracking an app, and marks it as updated
+// if the app is already tracked.
+func (mgr *Manager) Track(appRoot string) (*Instance, error) {
+	app, err := mgr.resolve(appRoot)
+	if err != nil {
+		return nil, err
+	}
+	_, err = mgr.db.Exec(`
```

Comment:
What if there are multiple checkout's of the same application?

Reply:
The primary key is the `appRoot` so they'll be stored multiple times. I went back and forth on this, not sure what's best exactly, but this approach works well for our current use case